### ISSUE TITLE
Bug "Undefined offset: 0" introduced in 1.12

### ISSUE THREAD

**Issue by ACTINEO-Ben** [doctrine/annotations#405]
There seems to be a new bug introduced with the latest 1.12 version.
Since I did not use it directly but used [l5-swagger](https://github.com/DarkaOnLine/L5-Swagger), which is using [swagger-php](https://github.com/zircote/swagger-php) which uses doctrine/annotations, I'm not exactly sure how to reproduce this easily.

What I can tell is that in [line 1447 of the DocParser.php](https://github.com/doctrine/annotations/compare/1.11.2...1.12.0#diff-e5d2e0837a5b53f6d38f5a1f6d4f154c757e2cba3eb3259751ed19944339a6edR1447) index "0" of the variable `$positionalArguments` is directly referenced, but the first set index in that variable at this point for me was "2".
This is what I saw when I `var_export`ed the content right before it failed:
![image](https://user-images.githubusercontent.com/77456352/108611116-9af96600-73db-11eb-8bbb-f3c0694f7a64.png)

So this should be an easy fix I think, but I'm not deep enough in that library to be able to both create a test and fix this issue myself.

For now I reverted back to 1.11.2 and everything is working again.

**Comment by kevinpapst** [comment]
The commit https://github.com/doctrine/annotations/commit/f1f293ade13b63c6820d2d37bcdb84bbd2e00ce9 introduced that regression/bug. It is happening for me as well.

This code here directly access index 0:

https://github.com/doctrine/annotations/blob/ebec9b1708c95d7602245c8172773f6b4e0c3be1/lib/Doctrine/Common/Annotations/DocParser.php#L1446-L1450

I have seen at least index 1 and 2, while 0 was not set.
For me it happens in on PHP 7.4, a Symfony project and it triggers in FosRestBundle controller with Swagger annotations.

Using this code instead fixed it for me:
```php
 if (count($positionalArguments) === 1) { 
     $value = array_pop($positionalArguments); 
 } else { 
     $value = array_values($positionalArguments); 
 } 
```

@Vincz can you remember if there was a reason accessing index 0?

**Comment by Vincz** [comment]
@kevinpapst My mistake. I wrongly assumed that positional arguments after named one was not valid. 
The problem should be fixed in #406

**Comment by kevinpapst** [comment]
Thanks for the late night fix!
